### PR TITLE
Add `dropout` argument to `MultiheadAttentionBlock`

### PR DESCRIPTION
**Pull request by rusty1s**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
 - Fix the default arguments of `DataParallel` class ([#6376](https://github.com/pyg-team/pytorch_geometric/pull/6376))
 - Fix `ImbalancedSampler` on sliced `InMemoryDataset` ([#6374](https://github.com/pyg-team/pytorch_geometric/pull/6374))
-- Breaking Change: Changed the interface and implementation of `GraphMultisetTransformer` ([#6343](https://github.com/pyg-team/pytorch_geometric/pull/6343))
+- Breaking Change: Changed the interface and implementation of `GraphMultisetTransformer` ([#6343](https://github.com/pyg-team/pytorch_geometric/pull/6343), [#6484](https://github.com/pyg-team/pytorch_geometric/pull/6484))
```
Comment (wsad1) [inline CHANGELOG.md:68]: Nitpick: Add a new changelog entry, since this PR doesn't break the current interface.

File: test/nn/aggr/test_gmt.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def test_graph_multiset_transformer():
     aggr = GraphMultisetTransformer(16, k=2, heads=2)
     aggr.reset_parameters()
     assert str(aggr) == ('GraphMultisetTransformer(16, k=2, heads=2, '
-                         'layer_norm=False)')
+                         'layer_norm=False, dropout=0.0)')
```
Comment (wsad1) [inline test/nn/aggr/test_gmt.py:14]: Add a test with non-zero dropout.

**Review by wsad1 — APPROVED** [review]
